src/command/restore/restore.c

```diff
@@ -2262,6 +2262,11 @@ cmdRestore(void)
         // Clean the data directory and build path/link structure
         restoreCleanBuild(jobData.manifest);
 
+        // Remove stanza archive spool path so existing files do not interfere with the new cluster. For instance, old archive-push
+        // acknowledgements could cause a new cluster to skip archiving. This should not happen if a new timeline is selected but
+        // better to be safe. Missing stanza spool paths are ignored.
+        storagePathRemoveP(storageSpoolWrite(), STORAGE_SPOOL_ARCHIVE_STR, .recurse = true);
```

Comment:
Hmm, good question. If the dir is missing there won't be an error, but if the dir exists and has bad permissions then likely archive-get will fail anyway, and that error will be in the PG logs where it is harder to see. But it's possible for `spool-path` to be configured/exist but `archive-async=n` in which case restore failing would be an issue.

Since we know nobody looks at warnings, and this will likely error eventually, might be best to leave it as a hard error. Thoughts, @sfrost? Maybe we should warn if `spool-path` is default and error if it is explicitly set?

Reply:
Would there be a situation where we use `archive-async=y` and don't have enough permission on the `spool-path` ? archive-get/push would fail. And indeed, warnings in PG logs are hard to find for beginners. We often have to specify to look at those logs for troubleshooting archiving issues...
So, imho, hard error is better.
However, if we're not using async archiving, we shouldn't fail on restore. Perhaps skipping this step if `archive-async=n` would be enough then ?

Reply:
Stephen agrees that a hard error is a good idea here, so let's go with that. However, it would be better if the error happened fast so I'm going to move it up to right after the PGDATA check.